release.sh

```diff
@@ -11,9 +11,18 @@ if ! env | grep GITHUB_TOKEN= > /dev/null; then
   exit 1
 fi
 
+# Run git fetch to ensure that the origin is updated.
+git fetch
```

Comment:
I think we should revert this change. I typically do a fetch and then do some testing etc before running the `make release` script. I would rather the script not auto update, if something lands in master maint during that local testing, I am fine if it waits till next release.

Reply:
This is to ensure that `git merge-base --is-ancestor` works correctly. It won't change your local branch. If you don't do this fetch, there's a larger chance that the below push will fail after you have already tagged it.

It's minimal since this script shouldn't be run very frequently anyway, but it exists as a potential problem.

It's not a pull so it won't affect anything local on your machine.

Reply:
OK yeah that makes sense.